hop/auth.py

```diff
@@ -30,14 +32,20 @@ class Auth(auth.SASLAuth):
         to the certificate.
     """
 
-    def __init__(self, user, password, ssl=True, method=SASLMethod.SCRAM_SHA_512, **kwargs):
+    def __init__(self, user, password, host="", ssl=True, method=SASLMethod.SCRAM_SHA_512,
```

Comment:
Does a host matching the empty string do anything special? I've seen that used in a few places.

Reply:
It does. `select_matching_auth` will consider a credential with an empty hostname a possible, but inexact match for any hostname. Such a credential will be selected if there is no truly exact match (and there is only one such inexact match).

Reply:
Thanks for the explanation, that makes sense.